llama-index-integrations/node_parser/llama-index-node-parser-relational-dashscope/llama_index/node_parser/dashscope/base.py

```diff
@@ -35,6 +35,10 @@ class DashScopeJsonNodeParser(BaseElementNodeParser):
         default="cn",
         description="language of tokenizor, accept cn, en, any. Notice that <any> mode will be slow.",
     )
+    _dashscope_api_key: str = PrivateAttr()
+
+    def __init__(self, dashscope_api_key: Optional[str] = None):
+        self._dashscope_api_key = dashscope_api_key or os.environ.get("DASHSCOPE_API_KEY")
```

Comment:
I'm almost certain here that Pydantic will complain if we don't call `super().__init__(...)` with the correct params.

You can follow the example I previous linked again for this: https://github.com/run-llama/llama_index/blob/f1965dfad3fbf9ff1e97e1948a4ed8604c9c266b/llama-index-integrations/llms/llama-index-llms-deepinfra/llama_index/llms/deepinfra/base.py#L134